freqtrade/resolvers/exchange_resolver.py

```diff
@@ -28,6 +28,7 @@ def __init__(self, exchange_name: str, config: dict) -> None:
         except ImportError:
             logger.info(
                 f"No {exchange_name} specific subclass found. Using the generic class instead.")
+        if not hasattr(self, "exchange"):
```

Comment:
in other places in the codebase another kind of checking (`if not edge`) is used... I propose to unify these checks.

Reply:
that does not work unfortunately (and the case is quite different here).

`self.exchange` does not yet exist in case of the import-error - and the only way to not have that fail is using hasattr, since we'd be checking a property which does not yet exist.

Initializing exchange to "None" would allow to check `if self.exchange` - but has the negative side-effect that the type changes from `Exchange` to `Optional[Exchange]` - which makes usage of this property a mess, since it will always exist after the __init__ function.

Initializing exchange within the exception handler does work (and prevents having to check) - but raises a very ugly exception (stacktrace) since it's raised within the exception handler so python thinks the original error is the cause.

`hasattr` is fine to check (but has performance implications, so should only be used when we know how often it's called (once - in this case).
Edge on the other hand is something optional - we're checking that often during regular operations.

Reply:
You are right.

LGTM 👍